src/Senders/CurlSender.php

```diff
@@ -84,4 +135,36 @@ public function setCurlOptions($handle, $scrubbedPayload, $accessToken)
             }
         }
     }
+
+    private function checkForCompletedRequests($accessToken)
+    {
+        do {
+            $curlResponse = curl_multi_exec($this->multiHandle, $active);
+        } while ($curlResponse == CURLM_CALL_MULTI_PERFORM);
+        while ($active && $curlResponse == CURLM_OK) {
```

Comment:
Actually, reading this a bit closer it seems like this should be broken up into different functions.